src/components/Topology/Topology.js

```diff
@@ -205,24 +282,27 @@ export default ({ modelData, width, height }) => {
       .attr("stroke-width", 3)
       .attr("stroke", "#888888")
       .call((_) => {
-        // When ever a new element is added zoom the canvas to fit.
-        const {
-          width: svgWidth,
-          height: svgHeight,
-        } = topo.node().getBoundingClientRect();
+        const topoNode = topo.node();
+        const { width: svgWidth, height: svgHeight } = topoNode
+          ? topoNode.getBoundingClientRect()
+          : {};
+
+        // Whenever a new element is added zoom the canvas to fit.
         if (svgWidth > 0 && svgHeight > 0) {
-          // Magic number that presents reasonable padding around the viz.
-          const padding = 200;
-          const scale = Math.min(
-            width / (svgWidth + padding),
-            height / (svgHeight + padding)
-          );
-          const translateX = (width - svgWidth * scale) / 2;
-          const translateY = (height - svgHeight * scale) / 2;
-          topo.attr(
-            "transform",
-            `translate(${translateX},${translateY}) scale(${scale},${scale})`
-          );
+          const scale = Math.min(width / svgWidth, height / svgHeight) || 1;
+
+          const translateX = (width - svgWidth * scale) / 2 || 0;
+          const translateY = (height - svgHeight * scale) / 2 || 0;
+          topo
+            .attr(
+              "transform",
+              `translate(${translateX},${translateY}) scale(${scale},${scale})`
+            )
+            .attr("width", `${width}`)
+            .call(
+              zoom.transform,
```

Comment:
Can you create an issues to track the... 
 - 'zoom in on click' issue when the topology is in fullscreen mode.
 - the perpetually smaller zoom level on the small topology after interacting with the full screen mode.

These also only seem to happen when applications already have existing positional annotations. For applications that are being auto-placed these issues don't appear to exist.